gaiacli stake validator gives :panic: UnmarshalBinary cannot decode empty bytes

gaiacli stake validator cosmosaccaddr1mtza07sz00578srn47new4wxyhefsv7wu5nn3h
panic: UnmarshalBinary cannot decode empty bytes


ecode empty bytes

goroutine 1 [running]:
github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/go-amino.(*Codec).MustUnmarshalBinary(0xc420090720, 0x0, 0x0, 0x0, 0x18438c0, 0xc420af2280)
	/Users/directory/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/go-amino/amino.go:253 +0x8a
github.com/cosmos/cosmos-sdk/x/stake/client/cli.GetCmdQueryValidator.func1(0xc420a58a00, 0xc420a47060, 0x1, 0x1, 0x0, 0x0)
	/Users/directory/go/src/github.com/cosmos/cosmos-sdk/x/stake/client/cli/query.go:35 +0x246
github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra.(*Command).execute(0xc420a58a00, 0xc420a47030, 0x1, 0x1, 0xc420a58a00, 0xc420a47030)
	/Users/directory/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra/command.go:762 +0x468
github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x1e801a0, 0x1895ae1, 0x24, 0xc420037288)
	/Users/directory/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra.(*Command).Execute(0x1e801a0, 0xc420a36d80, 0x1e80290)
	/Users/directory/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tmlibs/cli.Executor.Execute(0x1e801a0, 0x19d6550, 0x2, 0xc420a3eca0)
	/Users/directory/go/src/github.com/cosmos/cosmos-sdk/vendor/github.com/tendermint/tmlibs/cli/setup.go:89 +0x4e
main.main()
	/Users/directory/go/src/github.com/cosmos/cosmos-sdk/cmd/gaia/cmd/gaiacli/main.go:123 +0x8a7

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -101,6 +101,7 @@ BUG FIXES
 * \#1258 - printing big.rat's can no longer overflow int64
 * \#887  - limit the size of rationals that can be passed in from user input
 * \#1461 - CLI tests now no longer reset your local environment data
+* \#1505 - `gaiacli stake validator` no longer panics if validator doesn't exist
 
 ## 0.19.0
 

--- a/x/stake/client/cli/query.go
+++ b/x/stake/client/cli/query.go
@@ -31,8 +31,9 @@ func GetCmdQueryValidator(storeName string, cdc *wire.Codec) *cobra.Command {
 			res, err := ctx.QueryStore(key, storeName)
 			if err != nil {
 				return err
+			} else if len(res) == 0 {
+				return fmt.Errorf("No validator found with address %s", args[0])
 			}
-
 			validator := types.MustUnmarshalValidator(cdc, addr, res)
 
 			switch viper.Get(cli.OutputFlag) {

